pkg/api/v1/summary_test.go

```diff
@@ -181,3 +181,167 @@ func TestListTabSummaries(t *testing.T) {
 	}
 
 }
+
+func GetTabSummary(t *testing.T) {
+	tests := []struct {
+		name        string
+		config      map[string]*configpb.Configuration
+		summaries   map[string]*summarypb.DashboardSummary
+		req         *apipb.GetTabSummaryRequest
+		want        *apipb.GetTabSummaryResponse
+		expectError bool
+	}{
+		{
+			name: "Returns an error when there's no dashboard in config",
+			config: map[string]*configpb.Configuration{
+				"gs://default/config": {},
+			},
+			req: &apipb.GetTabSummaryRequest{
+				Dashboard: "missing",
+				Tab:       "Carpe Noctem",
+			},
+			expectError: true,
+		},
+		{
+			name: "Returns an error when there's no tab in config",
+			config: map[string]*configpb.Configuration{
+				"gs://default/config": {
+					Dashboards: []*configpb.Dashboard{
+						{
+							Name: "Aurora",
+							DashboardTab: []*configpb.DashboardTab{
+								{
+									Name: "Borealis",
+								},
+							},
+						},
+					},
+				},
+			},
+			req: &apipb.GetTabSummaryRequest{
+				Dashboard: "Aurora",
+				Tab:       "Noctem",
+			},
+			expectError: true,
+		},
+		{
+			name: "Returns an error when there's no summary for dashboard yet",
+			config: map[string]*configpb.Configuration{
+				"gs://default/config": {
+					Dashboards: []*configpb.Dashboard{
+						{
+							Name: "ACME",
+							DashboardTab: []*configpb.DashboardTab{
+								{
+									Name:          "me-me",
+									TestGroupName: "testgroupname",
+								},
+							},
+						},
+					},
+				},
+			},
+			req: &apipb.GetTabSummaryRequest{
+				Dashboard: "acme",
+				Tab:       "me-me",
+			},
+			expectError: true,
+		},
+		{
+			name: "Returns correct tab summary for a dashboard-tab",
```

Comment:
nit: I think there's one more case to add (summaries exist for tabs, but you request a nonexistent tab)